demos/sample_azure_iot_pnp/sample_azure_iot_pnp.h

```diff
@@ -0,0 +1,107 @@
+     * described in the corresponding DTMI. Should you choose to program your own PnP capable device,
+     * the functionality would need to match the DTMI and you would need to update the below 'model_id'.
+     * Please see the sample README for more information on this DTMI.
+     *
+     */
+    #define sampleazureiotMODEL_ID                                "dtmi:com:example:Thermostat;1"
+#endif
+
+extern AzureIoTHubClient_t xAzureIoTHubClient;
+
+/**
+ * @brief Provides the payload to be sent as telemetry to the Azure IoT Hub.
+ *
+ * @remark This function must be implemented by the specific sample.
+ *         `ulCreateTelemetry` is called periodically by the sample core task (the task created by `vStartDemoTask`). 
+ *         If the return is zero, telemetry is not sent to the Azure IoT Hub. 
+ * 
+ * @param[out] pucTelemetryData        Pointer to uint8_t* that will contain the Telemetry payload.
+ * @param[in]  ulTelemetryDataSize     Size of `pucTelemetryData`
+ * @param[out]  pulTelemetryDataLength The number of bytes written in `pucTelemetryData`
+ * 
+ * @return uint32_t Zero if successful, non-zero if any failure occurs.
+ */
+uint32_t ulCreateTelemetry( uint8_t * pucTelemetryData,
+                            uint32_t ulTelemetryDataSize,
+                            uint32_t * pulTelemetryDataLength );
+
+/**
+ * @brief Provides the payload to be sent as reported properties update to the Azure IoT Hub.
+ *
+ * @remark This function must be implemented by the specific sample.
+ *         `ulCreateReportedPropertiesUpdate` is called periodically by the sample 
+ *         core task (the task created by `vStartDemoTask`).
+ *         If the sample does not have any properties to update, just return zero to inform no 
+ *         update should be sent.
+ * 
+ * @param[out] pucPropertiesData    Pointer to uint8_t* that will contain the reported properties payload.
+ * @param[in]  ulPropertiesDataSize Size of `pucPropertiesData`
+ * 
+ * @return uint32_t The number of bytes written in `pucPropertiesData`.
+ */
+uint32_t ulCreateReportedPropertiesUpdate( uint8_t * pucPropertiesData,
+                                           uint32_t ulPropertiesDataSize );
+
+/**
+ * @brief Handles a Command received from the Azure IoT Hub.
+ * 
+ * @remark This function must be implemented by the specific sample.
+ *
+ * @param[in]  pxMessage                          Pointer to a structure that holds details of the Command.
+ * @param[out] pulResponseStatus                  Status code to be sent as response for Command request.
+ * @param[out] pucCommandResponsePayloadBuffer    Buffer in which to write a payload for the Command response.
+ * @param[in]  ulCommandResponsePayloadBufferSize Total size of `ucCommandResponsePayloadBuffer`.
+ * 
+ * @return uint32_t Number of bytes written to `ucCommandResponsePayloadBuffer`.
+ */
+uint32_t ulHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,

```

Comment:
>ulHandleCommand [](http://example.com/codeflow?start=9&length=15)

sample_azure_iot_pnp.h is header file whose implemenation is sample_azure_iot_pnp.c? But it looks like 

sample_azure_iot_pnp.h is interface for data provider
like
sample_azure_iot_pnp_simulate.c
sample_azure_iot_pnp_esp32_azure_iot_kit.c

Can we name this sample_azure_iot_pnp_device.h?

Reply:
We addressed this offline by defining the new names:

rename sample_azure_iot_pnp_data.h -> sample_azure_iot_pnp_data_if.h [DONE]
rename sample_azure_iot_pnp_device.c -> sample_azure_iot_pnp_simulated_data.c [DONE]
rename azure_iot_freertos_esp32_sensors -> azure_iot_freertos_esp32_sensors_data [DONE]